### PR TITLE
feat(1640): optionally display the STT field based on user email address

### LINKED ISSUE #1640: As an ACF user, I should not be required to select an STT on the request form
**Description:**
During testing for #1387, it was discovered that the request form forces the requestor to select an STT. This is only relevant for users in the states, territories, and tribes. This implicates that a system administrator will have to de-select location type when assigning permissions to non-STT users (e.g. OFA regional staff).

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] ACF users should **not** see STT option on request form
- [ ] Non-ACF users' form should remain unchanged
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] For the request form, optionally display the STT field based on user email address
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**

- Request form could vary based on the authentication method or email address. For example, if an ACF user, the field is region; else, STT.

### PR DESCRIPTION
**Pull request by jorgegonzalez**
## Summary of Changes
_Provide a brief summary of changes_

Pull request closes #1640

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
> *Demo GIF(s) and screenshots for testing procedure*

## Tabletop

- [x] Ready for Tabletop (Reviewer check this)
- [x] Tabletop scheduled (Assignee check this)
- [ ] Tabletop approval 

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [ ] ACF users should **not** see STT option on request form
- [ ] Non-ACF users' form should remain unchanged
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by riatzukiza — APPROVED** [review]

**Comment by jorgegonzalez** [comment]
I'm working on getting the tests to pass for this PR

**Review by abottoms-coder — APPROVED** [review]

File: tdrs-frontend/src/components/Home/Home.test.jsx [hunk 1]
```diff
@@ -302,6 +302,30 @@ describe('Pre-approval Home page', () => {
     expect(options.length).toEqual(4)
   })
 
+  it('should not show the stt combno box for non-STT users', () => {
```
Comment (ADPennington) [inline tdrs-frontend/src/components/Home/Home.test.jsx:305]: @jorgegonzalez is [this](https://github.com/raft-tech/TANF-app/blob/4a75b9d68fabb623ee42badc7da87a0ae9349ca6/tdrs-frontend/src/components/Home/Home.jsx#L170) check still happening? looks like i cant submit a request without completing the invisible fields on this form. 
1640_raft

Comment (jorgegonzalez) [reply tdrs-frontend/src/components/Home/Home.test.jsx:305]: Can you try again with the new changes?

**Review by ADPennington — APPROVED** [review]
thanks for the fixing the form validation @jorgegonzalez 🚀 tested the new acf user and new non-acf user route and can confirm that:
1.  acf users do not see the STT field and can submit the request
2. non-acf users see the stt field and can submit the request.

**Pull request merged**